crates/oxc_transformer/src/es2016/exponentiation_operator.rs

```diff
@@ -1,41 +1,257 @@
+    /// For `obj[ref]`, change it to `var _obj, _ref; _obj = obj, _ref = ref, _obj[_ref] = Math.pow(_obj[_ref], 2);`.
+    fn explode(
+        &mut self,
+        node: AssignmentTarget<'a>,
+        nodes: &mut Vec<'a, Expression<'a>>,
+    ) -> Option<Exploded<'a>> {
+        let node = match node {
+            AssignmentTarget::SimpleAssignmentTarget(target) => target,
+            AssignmentTarget::AssignmentTargetPattern(_) => {
+                // Invalid Syntax
+                return None;
+            }
+        };
+        let obj = self.get_obj_ref(self.ast.copy(&node), nodes)?;
+        let (reference, uid) = match node {
+            SimpleAssignmentTarget::AssignmentTargetIdentifier(ident) => {
+                let reference = AssignmentTarget::SimpleAssignmentTarget(
+                    SimpleAssignmentTarget::AssignmentTargetIdentifier(ident),
+                );
+                (reference, obj)
+            }
+            SimpleAssignmentTarget::MemberAssignmentTarget(member_expr) => {
+                let computed = member_expr.is_computed();
+                let prop = self.get_prop_ref(member_expr.unbox(), nodes)?;
+                let optional = false;
+                let obj_clone = self.ast.copy(&obj);
+                let span = Span::default();
+                let (reference, uid) = match &prop {
+                    Expression::Identifier(ident) if !computed => {
+                        let ident = IdentifierName::new(span, ident.name.clone());
+                        (
+                            self.ast.static_member(span, obj_clone, ident.clone(), optional),
+                            self.ast.static_member_expression(span, obj, ident, optional),
+                        )
+                    }
+                    _ => {
+                        let prop_clone = self.ast.copy(&prop);
+                        (
+                            self.ast.computed_member(span, obj_clone, prop_clone, optional),
+                            self.ast.computed_member_expression(span, obj, prop, optional),
+                        )
+                    }
+                };
+                (
+                    AssignmentTarget::SimpleAssignmentTarget(
+                        self.ast.simple_assignment_target_member_expression(reference),
+                    ),
+                    uid,
+                )
+            }
+            _ => return None,
+        };
+        Some(Exploded { reference, uid })
+    }
+
+    /// Make sure side-effects of evaluating `obj` of `obj.ref` and `obj[ref]` only happen once.
+    fn get_obj_ref(
+        &mut self,
+        node: SimpleAssignmentTarget<'a>,
+        nodes: &mut Vec<'a, Expression<'a>>,
+    ) -> Option<Expression<'a>> {
+        let reference = match node {
+            SimpleAssignmentTarget::AssignmentTargetIdentifier(ident) => {
+                if ident
+                    .reference_id
+                    .get()
```

Comment:
- [ ] Other places are using `clone().into_inner()`, we should change the other places.